test/e2e/fixture/fixture.go

```diff
@@ -123,10 +123,18 @@ func EnsureCleanState() {
 	start := time.Now()
 
 	// delete resources
-	FailOnErr(Run("", "kubectl", "-n", ArgoCDNamespace, "delete", "app", "--all"))
+
+	text, err := Run("", "kubectl", "get", "app", "-o", "name")
+	CheckError(err)
+	for _, name := range strings.Split(text, "\n") {
+		if name != "" {
+			// is it much more reliable to get argocd to delete an app than kubectl
```

Comment:
Something for team discussion maybe.

Reply:
The `argocd app delete` will try to delete child resources. This will make tests much slower. I would delete apps using kubectl and delete deployment namespace in backgroup

Reply:
could just delete without finalizer?

Reply:
I'm ok with it. The `kubectl delete app --all` would do the same, but I'm fine if you want to use `argocd app delete`

Reply:
Let me gather data on this. I felt that deleting apps was fast (<1s),  whereas debugging Argo CD  when it could not understand where an app had gone was expensive.